src/docfx/build/Build.cs

```diff
@@ -226,11 +222,35 @@ private static string GetOutputPath(Document file, List<string> monikers)
             return outputPath;
         }
 
-        private static ResourceModel BuildResource(Document file)
+        private static (List<Error> errors, object model, List<string> monikers) BuildAssert(Document file, ContentType contentType)
```

Comment:
how about changing this method into a generic one:
```csharp
private static (List<Error> errors, T model, List<string> monikers) BuildAssert<T>(Document file, ContentType contentType, Func<Document, List<string>, T> func)
```

Reply:
Can we try not to merge `Redirection` with `Resource` ? Yes they currently looks almost the same, but they could be very different like `Resource` needs to be copied to output etc.

Reply:
The upper proposal is just to avoid the switch case based on type, which is the generic method invented for.
I am not sure how they will look later, although like a copying `Resource` to output can be handled outside this `BuildAsset` method

Reply:
The concept `Asset` (not Assert) here is misleading, is seems to be _Resource_ + _Redirection_, which used to refer to `Resource`. It is a new term (only used here) for people to grasp.

I really want to see methods like `BuildXXX` where `XXX` is the value defined in `ContentType` for readability. Inside these methods, there are numerous ways to share code, but please keep the build structure as is.

Reply:
that's a valid point, keep them separated then